feat(#8226): document how to hide fields/groups in forms (#2208)
medic/cht-docs: content/en/building/forms/configuring/hiding-form-content.md

---
title: "Hiding Form Content"
linkTitle: "Hiding Form Content"
weight: 4
description: >
  Controlling visibility of questions within a form
relatedContent: >
  building/forms
  building/forms/configuring/form-inputs
---

There are several different ways to control visibility of questions and other content within a form.  

## Conditionally hiding fields

> Expressions in the [`relevant` column](https://docs.getodk.org/form-logic/#conditionally-showing-questions) can be used to dynamically control the visibility of questions, groups, etc. When the expression evaluates to `true`, the content is shown; when it evaluates to `false`, the content is hidden. This allows for showing/hiding content based on previous answers or other data in the form. The relevancy of a group is inherited by the fields it contains. So, when a group is non-relevant, all of its contents will also be considered non-relevant.

Reviewer: ```suggestion
Expressions in the `relevant` [column](https://docs.getodk.org/form-logic/#conditionally-showing-questions) can be used to dynamically control the visibility of questions, groups, etc. When the expression evaluates to `true`, the content is shown; when it evaluates to `false`, the content is hidden. This allows for showing/hiding content based on previous answers or other data in the form. The relevancy of a group is inherited by the fields it contains so, when a group is non-relevant, all of its contents will also be considered non-relevant.
```

When the form is completed, values for non-relevant fields will _NOT be stored in the database._

## Fields hidden by default

> [Several types of fields](https://docs.getodk.org/form-question-types/#hidden-questions) are never visible to the user. 

Reviewer: ```suggestion
For a complete list of hidden fields, see [ODKs site](https://docs.getodk.org/form-question-types/#hidden-questions), otherwise see CHT relevant fields below. 
```

### `calculate` type

> [`calculate` fields](https://docs.getodk.org/form-logic/#calculations) are useful for storing data the user does not need to see but are referenced/used elsewhere in the form. The value contained in a `calcuate` field will be stored in the database when the form is completed.

Reviewer: ```suggestion
`calculate` [fields](https://docs.getodk.org/form-logic/#calculations) are useful for storing data the user does not need to see but are referenced/used elsewhere in the form. The value contained in a `calcuate` field will be stored in the database when the form is completed.
```
